pyomo/contrib/doe/doe.py

```diff
@@ -1062,13 +1171,10 @@ def _generate_scenario_blocks(self, model=None):
             model.scenarios = range(len(model.base_model.unknown_parameters) + 1)
         else:
             raise AttributeError(
```

Comment:
You may want to have this be a `DeveloperError` or `MouseTrap` error instead of an `AttributeError`.